oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -606,11 +609,11 @@ impl RuntimeRef {
         // to do that we first need to provide a reference to the caller node as a parameter to this
         // function.
 
-        let mut nodes = self.nodes.write().unwrap();
         let reference = self.new_node_reference();
 
         let reader = self.channels.duplicate_reference(reader)?;
 
+        let mut nodes = self.nodes.write().unwrap();
```

Comment:
Couldn't this go even further down? Perhaps add a comment either way.

Reply:
Might also be better to combine the `write` and `insert` into an `add_running_node()` method on `Runtime` or `RuntimeRef`, so the locking scope is small and clear.

Reply:
It could have gone further down you are right. I have moved to a dedicated function as per @daviddrysdale